server/src/packages/landuse-form/landuse-actions/landuse-actions.controller.ts

```diff
@@ -29,16 +29,35 @@ export class LanduseActionsController {
   @Patch('/:id')
   async update(@Body() body, @Param('id') id) {
     const allowedAttrs = pick(body, LANDUSE_ACTION_ATTRS);
+    const dateOfPreviousApproval = await this.setDate(body.dcp_dateofpreviousapproval);
+    const lapseDateOfPreviousApproval = await this.setDate(body.dcp_lapsedateofpreviousapproval);
+    const recordationDate = await this.setDate(body.dcp_recordationdate);
 
     await this.crmService.update('dcp_landuseactions', id, {
       ...allowedAttrs,
 
       ...(body.chosen_zoning_resolution_id ? { 'dcp_zoningresolutionsectionactionispursuantto@odata.bind': `/dcp_zoningresolutions(${body.chosen_zoning_resolution_id})` } : {}),
+      dcp_dateofpreviousapproval: dateOfPreviousApproval,
```

Comment:
Relating to [Comment A] below, I think here these fields should only be set on the update object if the values aren't null.

Here it looks like you would still send an update request with `null` assigned to these date properties.

Reply:
I think this goes back to our conversation about whether we should make just the **field** conditional OR the **entire update object**. I tried to make it as simple as possible by just doing the field, but maybe doing the entire object is safe in the long run

Reply:
Sounds good. You put that very clearly and it sounds like you're aware of the Pros, Cons, dangers and tradeoffs. So we're making a calculated decision here. Thanks!